### PR TITLE
[WFLY-11868] RESTEasy integration with Elytron on the client side

### PR DESCRIPTION
**Pull request by Skyllarr**
JIRA: https://issues.jboss.org/browse/WFLY-11868

### CODE REVIEW

File: elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11697[WFLY-11868]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1097[EAP7-1219]
```
Comment (asoldano) [inline elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc]: Please fix the link, right one is likely https://issues.jboss.org/browse/EAP7-1219

File: elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Wildfly-config.xml file is used by Elytron client and it contains the authentication configuration that should be used
+when attempting to connect to the server (username, password, ssl config, mechanism specification etc).
+This requested feature is about making use of Elytron and wildfly-config.xml file on the client side of REST deployment.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11697[WFLY-11868]
```
Comment (asoldano) [inline elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc]: Please fix the link, right one is likely https://issues.jboss.org/browse/WFLY-11868

File: elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* TBD
+
+=== Affected Projects or Components
+
+* Security
+* Web Services
```
Comment (asoldano) [inline elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc]: Please replace Web Services with REST / RESTEasy

**Comment by Skyllarr** [comment]
@asoldano I have fixed the issues you mentioned. I'll keep the backward compatibility in mind, thank you for the feedback.

**Comment by marekkopecky** [comment]
@Skyllarr With current RESTEasy client approach, I can have two clients with different credentials and I can use these both clients for the requests for one server. Is this possible with wildfly-config.xml approach?

File: elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+* https://issues.jboss.org/browse/EAP7-1219[EAP7-1219]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* TBD
```
Comment (marekkopecky) [inline elytron/WFLY-11868_RESTEasy_client_integration_with_Elytron.adoc]: It's me -> mkopecky@redhat.com

**Comment by marekkopecky** [comment]
`Testing By` section is missing.

**Comment by Skyllarr** [comment]
> @Skyllarr With current RESTEasy client approach, I can have two clients with different credentials and I can use these both clients for the requests for one server. Will this be possible with wildfly-config.xml approach?

@marekkopecky wildfly-config.xml can have multiple configurations (with different credentials) defined. Rules are then applied to each request to conclude which config to use, and the rule can take eg. username into account as well ([relevant wildfly-config doc section](https://docs.jboss.org/author/display/WFLY/WildFly+Client+Configuration#WildFlyClientConfiguration-%3Cauthenticationrules%2F%3Eand%3Csslcontextrules%2F%3E)). I think this should be possible as long as there is a way to decide which client config to use when making a request.

**Comment by Skyllarr** [comment]
Hello @ronsigal @darranl @marekkopecky , I have updated the analysis document. Please review. Thanks in advance!

**Comment by Skyllarr** [comment]
> @Skyllarr With current RESTEasy client approach, I can have two clients with different credentials and I can use these both clients for the requests for one server. Will this be possible with wildfly-config.xml approach?

Hello @marekkopecky , I have read your question again and maybe can provide clearer answear. You can set different credentials in Elytron client for the same URL. But if there is no additional rule (eg. port) that specifies which credentials to choose ( I think is what you were asking), then the first credentials will be applied always (in order that they were configured). But I don't think this will result in backwards incompatibility, because user can specify their own credentials for every RESTEasy client like before. This will overwrite credentials that were set by Elytron. But I will keep this in mind during implementation.

**Pull request merged**